src/inmanta/agent/agent.py

```diff
@@ -114,7 +114,7 @@ def is_done(self) -> bool:
 
     def cancel(self) -> None:
         if not self.is_running() and not self.is_done():
-            LOGGER.info("Cancelled deploy of %s %s", self.gid, self.resource_id)
+            self.logger.info("Cancelled deploy of %s %s %s", self.__class__.__name__, self.gid, self.resource_id)
```

Comment:
Just for my information. Why is it important to rely on the specific logger from the agent?

Reply:
currently, we see the cancel being logged, but we don't know which agent instance produced it. 
So to analyze the agent getting stuck, it is near impossible to say if it goes into the cancel code path at all. 

(cross-agent-dependencies are also getting cancelled here, meaning that the agent name in the resource does not necessarily correspond to the agent it is on, so that means you have to make  assumptions, which is not very useful when diagnosing a seemingly impossible scenario, where you know some of your assumptions don't hold).